client/components/data/query-site-front-page/index.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * External dependencies
+ */
+
+import { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import getSiteSetting from 'state/selectors/get-site-setting';
+import { updateSiteSettings } from 'state/site-settings/actions';
+import { getSiteOption } from 'state/sites/selectors';
+
+class QuerySiteFrontPage extends Component {
+	static propTypes = {
+		siteId: PropTypes.number,
+	};
+
+	componentDidMount() {
+		if ( this.props.showOnFrontSetting ) {
```

Comment:
Let's avoid this as we might end up in a state where we have stale data. Query components are intended to always fetch at least once on mount.

Reply:
Let me know if caching wasn't the purpose here. I might be misreading this otherwise.